src/plonk/prover.rs

```diff
@@ -241,7 +245,7 @@ pub fn create_proof<C: CurveAffine, T: TranscriptWrite<C>, ConcreteCircuit: Circ
         .collect::<Result<Vec<_>, _>>()?;
 
     // Sample theta challenge for keeping lookup columns linearly independent
-    let theta = ChallengeTheta::get(transcript);
+    let theta = transcript.squeeze_challenge_scalar();
```

Comment:
```suggestion
    let theta: ChallengeTheta = transcript.squeeze_challenge_scalar();
```
Otherwise, we lose some of the advantage of the explicit typing, because the compiler could infer done other challenge type if we were to use `theta` in the wrong place. Similarly for the other challenges.

Reply:
I think the compiler is inferring the type `theta: ChallengeTheta` here from the signature of `lookup.commit_permuted()` (on line 260) below.

Reply:
It is, but we can't see that in the code of this function. If we called a variable `theta` but accidentally passed it to an API expecting `beta`, for example, then the compiler wouldn't catch that. It would just infer the type as `ChallengeBeta`. I want to be able to know *when reading the declaration* that the compiler would catch any attempt to pass this variable in the wrong place.